src/Lock.ts

```diff
@@ -31,9 +32,11 @@ export abstract class Lock {
     acquireAttemptsLimit = defaultTimeoutOptions.acquireAttemptsLimit,
     retryInterval = defaultTimeoutOptions.retryInterval,
     refreshInterval = Math.round(lockTimeout * REFRESH_INTERVAL_COEF),
-    onLockLost = defaultOnLockLost
+    onLockLost = defaultOnLockLost,
+    externallyAcquiredIdentifier
   }: LockOptions = defaultTimeoutOptions) {
-    this._identifier = uuid4()
+    this._identifier = externallyAcquiredIdentifier || uuid4()
```

Comment:
unrelated to this PR, but since you only support Node 14+, you can actually use native [randomUUID](https://nodejs.org/api/crypto.html#cryptorandomuuidoptions), which is significantly faster, and get rid of the dependency.